accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -118,7 +122,7 @@ The declaration introduces an extension. The extension's `on` type defines which
 
 For any member access, `x.foo`, `x.bar()`, `x.baz = 42`, `x(42)`, `x[0] = 1` or `x + y`, including null-aware and cascade accesses which effectively desugar to one of those direct accesses, and including implicit member accesses on `this`, the language first checks whether the static type of `x` has a member with the same base name as the operation. That is, if it has a corresponding instance member, respectively, a `foo` method or getter or a `foo=` setter. a `bar` member or `bar=` setter, a `baz` member or `baz=` setter, a `call` method, a `[]=` operator or a `+` operator. If so, then the operation is unaffected by extensions. *This check does not care whether the invocation is otherwise correct, based on number or type of the arguments, it only checks whether there is a member at all.*
 
-(The types `dynamic` and `Never` are considered as having all members, the type `void` is always a compile-time error when used in a receiver position, so none of these can ever be affected by static extension methods. The type `Function` and all function types are considered as having a `call` member on top of any members inherited from `Object`. Methods declared on `Object` are available on all types and can therefore never be affected by extensions).
+(The types `dynamic` is considered as having all members, the type `void` and `Never` are always compile-time errors when used in a receiver position, so none of these can ever be affected by static extension methods. The type `Function` and all function types are considered as having a `call` member on top of any members inherited from `Object`. Methods declared on `Object` are available on all types and can therefore never be affected by extensions).
```

Comment:
Typo:

```suggestion
(The type `dynamic` is considered as having all members, the type `void` and `Never` are always compile-time errors when used in a receiver position, so none of these can ever be affected by static extension methods. The type `Function` and all function types are considered as having a `call` member on top of any members inherited from `Object`. Methods declared on `Object` are available on all types and can therefore never be affected by extensions).
```

Leaf wasn't quite happy about `are always compile-time errors` for `Never`. Maybe the text could simply avoid to state an answer to that question?

Reply:
True, it doesn't matter which behavior we choose for Never: Disallow any member access or allow any member access, because either will prevent extension members from applying. (Disallowing any member access is not the same as sayting that there isn't any member, it's a hard error whenever you try to do `.foo` on anything with a static type of `void` (and possibly `Never`).

Interestingly, we'll probaly *allow* `ExtName(neverExpression)`. There is no harm in that, it has a well-defined static behavior, and it will never get run at run-time anyway.

Reply:
Just noted one more thing: Leaf actually wrote 'It is an error to call a method, setter, or getter on a receiver of static type `Never` (including via a null aware operator).' in the NNBD feature spec.

Reply:
I'll keep it unspecified here since it won't make a difference for extension methods until we do get NNBD.